[api-minor] Remove SystemJS usage, in development mode, from the worker

Snuffleupagus opened the pull request:
Now that https://bugzilla.mozilla.org/show_bug.cgi?id=1247687 has landed in Firefox, we're able to use worker-modules during development :-)

This removes the final piece of SystemJS usage from the PDF.js library, thus allowing a fair bit of clean-up, and we now use *only* native `import`/`export` statements everywhere in development mode.

Snuffleupagus committed:
Remove SystemJS usage, in development mode, from the worker

Now that https://bugzilla.mozilla.org/show_bug.cgi?id=1247687 has landed in Firefox, we're able to use worker-modules during development :-)

This removes the final piece of SystemJS usage from the PDF.js library, thus allowing a fair bit of clean-up, and we now use *only* native `import`/`export` statements everywhere in development mode.

Snuffleupagus committed:
Revert "Add a `getArrayLookupTableFactory` helper function and use it to re-format `src/core/{glyphlist, unicode}.js`"

This reverts commit 56fa6d414cb1115e03f9c1aa9f1d5bc52efcb7ac now that SystemJS is gone.

Snuffleupagus committed:
Introduce some optional chaining in the `src/shared/` folder

Snuffleupagus commented:
/botio integrationtest

Snuffleupagus committed:
Change the `createPromiseCapability` helper function into a `PromiseCapability` class

This is not only slightly more compact, but it also simplifies the handling of the `settled` getter.

Snuffleupagus committed:
Introduce some logical assignment in the `src/core/` folder

Snuffleupagus commented:
We probably need to wait a little while before landing this, in case of any follow-up work and/or back-outs, but things appear to work just fine locally with this patch-series.

E3qwe4w commented:
Я не понимаю Правда что это я знаю это приложение с кошкой читала не поняла
читала выборочно наткнулась когда пыталась найти свои токены вы просто
скажите что это и что мне с этим делать а я сейчас открыла и подумала что
кто-то имеет доступ к телефону аккаунту это канечно мне не нравится если
ещё знать что как и почему и коротко

чт, 15 дек. 2022 г., 18:24 Jonas Jenwald ***@***.***>:

> We probably need to wait a little while before landing this, in case of
> any follow-up work and/or back-outs, but things appear to work just fine
> locally with this patch-series.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/pdf.js/pull/12563#issuecomment-1353063495>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4T4LSBE7T4O522NCPSE4FDWNMLZTANCNFSM4THIO2BQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

E3qwe4w commented:
Почитала мельком это моя информация которую можно востоновить ? Сколько
стоит это ?

чт, 15 дек. 2022 г., 19:03 Людмила Петрова ***@***.***>:

> Я не понимаю Правда что это я знаю это приложение с кошкой читала не
> поняла читала выборочно наткнулась когда пыталась найти свои токены вы
> просто скажите что это и что мне с этим делать а я сейчас открыла и
> подумала что кто-то имеет доступ к телефону аккаунту это канечно мне не
> нравится если ещё знать что как и почему и коротко
>
> чт, 15 дек. 2022 г., 18:24 Jonas Jenwald ***@***.***>:
>
>> We probably need to wait a little while before landing this, in case of
>> any follow-up work and/or back-outs, but things appear to work just fine
>> locally with this patch-series.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/mozilla/pdf.js/pull/12563#issuecomment-1353063495>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A4T4LSBE7T4O522NCPSE4FDWNMLZTANCNFSM4THIO2BQ>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>

Snuffleupagus commented:
/botio test

E3qwe4w commented:
Вот вы пишите успешное слияние может закрыть эту тему вопрос слияние чего?
И какую тему? Вот вы можете понять что я не понимаю что приходится
додумывать нужно конкретно и коротко что вы предлагаете?  что от меня
требуется? Ещё зачем?

E3qwe4w commented:
Я делаю скиншоты потому что мечтаю что там кто смотрит за казено подумает
вдруг что я какой-нибудь человек который снимает репортаж и даст мне
выйграть для хорошего репортажа

пт, 16 дек. 2022 г., 15:11 Людмила Петрова ***@***.***>:

> Вот вы пишите успешное слияние может закрыть эту тему вопрос слияние чего?
> И какую тему? Вот вы можете понять что я не понимаю что приходится
> додумывать нужно конкретно и коротко что вы предлагаете?  что от меня
> требуется? Ещё зачем?
>

E3qwe4w commented:
На собрание нужно в школу потом в садик если будут письма прочитаю позже

пт, 16 дек. 2022 г., 15:15 Людмила Петрова ***@***.***>:

> Я делаю скиншоты потому что мечтаю что там кто смотрит за казено подумает
> вдруг что я какой-нибудь человек который снимает репортаж и даст мне
> выйграть для хорошего репортажа
>
> пт, 16 дек. 2022 г., 15:11 Людмила Петрова ***@***.***>:
>
>> Вот вы пишите успешное слияние может закрыть эту тему вопрос слияние
>> чего? И какую тему? Вот вы можете понять что я не понимаю что приходится
>> додумывать нужно конкретно и коротко что вы предлагаете?  что от меня
>> требуется? Ещё зачем?
>>
>

timvandermeij commented:
Nice to see movement on this! Sadly the upstream patch has been backed out, but hopefully that can be fixed soon and we can finally get rid of SystemJS completely :-)

Snuffleupagus commented:
[Bug 1247687](https://bugzilla.mozilla.org/show_bug.cgi?id=1247687) was re-landed a week ago, and it seems like it stuck this time :-)

The question now is how long we should wait before deeming it safe to land this PR; do we perhaps need to wait until worker-modules reach Firefox Beta [in about three weeks](https://wiki.mozilla.org/Release_Management/Calendar) or can we land this right away?

Snuffleupagus commented:
*Please note:* This PR only applies to the development viewer, i.e. `gulp server` mode, and won't affect e.g. the Firefox PDF Viewer.

---

https://bugzilla.mozilla.org/show_bug.cgi?id=1247687 was re-landed, and it seems like it stuck this time :-)

However, there's now also https://bugzilla.mozilla.org/show_bug.cgi?id=1812591 which tracks it reaching Firefox release and https://bugzilla.mozilla.org/show_bug.cgi?id=1812628 which will (I assume) add a new preference for worker modules.

Hence I wonder if we still need to wait for worker modules to officially ship in Firefox, or if we can land this PR now and assume that any PDF.js development is done using Firefox Nightly (possibly with a suitable browser preference set)?

/cc @calixteman, @timvandermeij Any opinions/thoughts on the above?

timvandermeij commented:
I would suggest waiting until it shipped in the stable version, at the very least until the preference is set unconditionally, to make sure we're not hindered by any bugs that may be found given that this is brand new functionality. It would be good to test PDF.js against Firefox Nightly to see if everything works as intended, but I'd feel much better about merging this once the upstream feature had some time to settle.

Snuffleupagus committed:
Simplify initialization of `static` class properties in the worker-thread

Now that we no longer depend on the old Babel version in SystemJS we can remove the `static get ...` work-arounds used to define constants, which leads to slightly more compact code.

Snuffleupagus commented:
This is now also blocked on https://bugzilla.mozilla.org/show_bug.cgi?id=1816933 being fixed.

Snuffleupagus committed:
Remove the closure from `BitModel` in the `src/core/jpx.js` file

Snuffleupagus commented:
https://bugzilla.mozilla.org/show_bug.cgi?id=1812591 has now landed in Firefox Nightly, see also https://groups.google.com/a/mozilla.org/g/dev-platform/c/J5zqdaNSpAM, which should've been the final blocker for this PR :-)

*Please note:* This patch only introduces worker modules for the development viewer, i.e. `gulp server`, which help improve the loading performance of http://localhost:8888/web/viewer.html

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio-linux preview

calixteman reviewed (approved):
It looks good overall.
It's good to see this patch becoming a reality.
Thank you.

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio test

timvandermeij commented:
Nice work, and really good to get rid of SystemJS now!
